skare3/dashboard/ska_github_info.py

```diff
@@ -0,0 +1,117 @@
+    tag_info = API.get(f'repos/{owner}/{repo}/git/ref/tags/{last_tag["tag_name"]}').json()
+    tag_sha = tag_info['object']['sha']
+    rel_commit = API.get(f'repos/{owner}/{repo}/commits/{tag_sha}').json()
+    commit_date = rel_commit['commit']['author']['date']
+
+    commits = API.get(f'repos/{owner}/{repo}/commits',
+                      params={'sha':'master', 'since': commit_date}).json()
+    commits = commits[:-1]  # remove the commit associated to the release
+
+    branches = API.get(f'repos/{owner}/{repo}/branches').json()
+    n_branches = len(branches)
+
+    issue_page = API.get(f'repos/{owner}/{repo}/issues', params={'per_page':100}).json()
+    issues = issue_page
+    while len(issue_page) == 100:
+        issue_page = API.get(f'repos/{owner}/{repo}/issues', params={'per_page':100}).json()
+        issues += issue_page
+    n_pr = len([i for i in issues if 'pull_request' in i])
+    n_issues = len(issues) - n_pr
+
+    merges = []
+    for commit in commits:
+        msg = commit['commit']['message']
+        match = re.match('Merge pull request (?P<pr>.+) from (?P<branch>\S+)\n\n(?P<description>.+)', msg)
+        if match:
+            msg = match.groupdict()
+            merges.append(f'PR{msg["pr"]}: {msg["description"]}')
+
+    repo_info = {
+        'owner': owner,
+        'name': repo,
+        'last_tag': last_tag["tag_name"],
+        'last_tag_date': last_tag['published_at'],
+        'commits': len(commits),
+        'merges': len(merges),
+        'merge_info': merges,
+        'issues': n_issues,
+        'pull_requests': n_pr,
+        'branches': n_branches
+    }
+    return repo_info
+
+
+def get_repositories_info(repositories):
+    info = []
+    for owner_repo in repositories:
+        print(owner_repo)
+        info.append(get_repository_info(owner_repo))
+    return info
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-o', default='repository_info.json')
+    args = parser.parse_args()
+
+    info = get_repositories_info([
+        'sot/Chandra.Maneuver',
+        'sot/Chandra.Time',
+        'sot/Quaternion',
+        'sot/Ska.DBI',
+        'sot/Ska.File',
+        'sot/Ska.Matplotlib',
+        'sot/Ska.Numpy',
+        'sot/Ska.ParseCM',
+        'sot/Ska.Shell',
+        'sot/Ska.Sun',
+        'sot/Ska.arc5gl',
+        'sot/Ska.astro',
+        'sot/Ska.ftp',
+        'sot/Ska.quatutil',
+        'sot/Ska.tdb',
+        'sot/acdc',
+        'sot/acis_taco',
+        'sot/agasc',
+        'sot/annie',
+        'sot/chandra_aca',
+        'sot/cmd_states',
+        'sot/cxotime',
+        'sot/eng_archive',
+        'sot/hopper',
+        'sot/kadi',
+        'sot/maude',
+        'sot/mica',
+        'sot/parse_cm',
+        'sot/proseco',
+        'sot/pyyaks',
+        'sot/ska_path',
+        'sot/ska_sync',
+        'sot/sparkles',
+        'sot/starcheck',
+        'sot/tables3_api',
+        'sot/testr',
+        'sot/xija'
```

Comment:
I think this is, offhand, missing find_attitude and ska_helpers to start.  Not sure if other not-ska3-flight not-Python packages like task_schedule and watch_cron_logs belong here too.  

And not sure if having the list hard-coded here is the best way to manage it?  But probably sufficient given how we develop packages.

Reply:
Seems like a dynamically-acquired list of every repo in the sot org might be the way to go.  One can apply filters, e.g. for a repo that has had no commits in the last N years, or whatever.  But in general I'm interested in seeing activity and status across the board.  This can point to repos that might be best "archived", who knows.

Reply:
Of course it might turn out that we lower the S/N of the table too much by having a bunch of inactive repos.  Not sure offhand.

Reply:
I was wondering if something like a Watched label could be used on the github side to make the list.

Reply:
We can have everything!

One could get all repositories of a given "owner" (in our case 'sot' or 'acis').

Reply:
that's the point of having filters on top. If the filters do not work for us, we make the list smaller.

Reply:
I like the owner idea.  In fact I bet @jzuhone would be happy to get this status dashboard for `acisops`.  (WIP at http://cxc.cfa.harvard.edu/mta/ASPECT/skare3/dashboard/)

Reply:
I think that filtering repos by age of last commit (which could be a new column in the table, in years) would be reasonable.  I'm not aware of any way to label a repository as a whole.  If it does exist, I would have the label opt out of being watched rather than opt in.